src/basic_memory/api/middleware/deprecation.py

```diff
@@ -0,0 +1,163 @@
+"""Deprecation middleware for v1 API endpoints.
+
+This middleware adds deprecation headers to v1 API responses and tracks
+usage metrics to help monitor the migration to v2.
+"""
+
+from collections import Counter
+from datetime import datetime, timedelta
+
+from fastapi import Request
+from loguru import logger
+from starlette.middleware.base import BaseHTTPMiddleware
+
+
+class DeprecationMetrics:
+    """Track v1 and v2 API usage for migration planning."""
+
+    def __init__(self):
+        """Initialize metrics counters."""
+        self.v1_calls = Counter()
+        self.v2_calls = Counter()
+
+    def record_v1_call(self, endpoint: str, client: str | None = None):
+        """Record a v1 API call.
+
+        Args:
+            endpoint: The endpoint path that was called
+            client: Optional client identifier
+        """
+        self.v1_calls[endpoint] += 1
+
+    def record_v2_call(self, endpoint: str):
+        """Record a v2 API call.
+
+        Args:
+            endpoint: The endpoint path that was called
+        """
+        self.v2_calls[endpoint] += 1
+
+    def get_stats(self) -> dict:
+        """Get usage statistics.
+
+        Returns:
+            Dictionary with v1/v2 call counts and adoption metrics
+        """
+        total_v1 = sum(self.v1_calls.values())
+        total_v2 = sum(self.v2_calls.values())
+        total = total_v1 + total_v2
+
+        return {
+            "v1_calls": total_v1,
+            "v2_calls": total_v2,
+            "total_calls": total,
+            "v2_adoption_rate": total_v2 / total if total > 0 else 0,
+            "top_v1_endpoints": self.v1_calls.most_common(10),
+            "top_v2_endpoints": self.v2_calls.most_common(10),
+        }
```

Comment:
if we are going to do metrics, I'd say use logfire. And I do want metrics, but I'm not sure we need these